lib/system-info.js

```diff
@@ -1,219 +1,296 @@
   } else if (platform.match(/bsd/i)) {
-    getSysctlValue(['hw.realmem'], function getMem(memory) {
-      callback(parseInt(memory, 10) / (1024 * 1024))
-    })
+    const memory = await getSysctlValue(['hw.realmem'])
+    return parseInt(memory, 10) / (1024 * 1024)
   } else if (platform.match(/linux/i)) {
-    readProc('/proc/meminfo', function parseProc(err, data) {
-      callback(parseMemInfo(data))
-    })
-  } else {
-    logger.debug('Unknown platform: %s; could not retrieve memory info', platform)
-    callback(null)
+    const data = await readProc('/proc/meminfo')
+    return parseMemInfo(data)
   }
+
+  logger.debug('Unknown platform: %s; could not retrieve memory info', platform)
+  return null
 }
 
-function getKernelVersion(callback) {
+/**
+ * Helper method for retrieving Kernel version information for different platforms
+ *
+ * @returns {*} null if unknown platform, otherwise string representation of kernel version
+ */
+async function getKernelVersion() {
   if (platform.match(/darwin/i) || platform.match(/bsd/i)) {
-    getSysctlValue(['kern.version'], function getMem(version) {
-      callback(version)
-    })
+    return await getSysctlValue(['kern.version'])
   } else if (platform.match(/linux/i)) {
-    readProc('/proc/version', function parseProc(err, data) {
-      callback(data)
-    })
-  } else {
-    logger.debug('Unknown platform: %s; could not read kernel version', platform)
-    callback(null)
+    return await readProc('/proc/version')
   }
+
+  logger.debug('Unknown platform: %s; could not read kernel version', platform)
+  return null
 }
 
-function getSysctlValue(names, callback) {
-  if (!names) {
-    return callback(null)
-  }
-  let returned = false
-  let ran = 0
-  names.forEach(function sysctlName(name) {
-    exec('sysctl -n ' + name, respond)
+/**
+ * Helper method for getting sysctl information given a list of potential values to look up
+ * Returns the first successful sysctl's output
+ *
+ * @param {Array.<string>} names List of sysctl values to look up
+ * @returns {*} null if we failed to lookup any info (error or not), or the first successful sysctl's output
+ */
+async function getSysctlValue(names = []) {
+  let returnValue = null
 
-    function respond(err, stdout, stderr) {
-      if (returned) {
-        return
-      }
-      if (err) {
-        logger.debug('Error when trying to run: sysctl -n %s: %s', name, err.message)
-        callback(null)
-        returned = true
-      } else if (!stderr) {
-        callback(stdout)
-        returned = true
-      }
-      if (++ran === names.length && !returned) {
-        logger.debug('No sysctl info found for names: %j', names)
-        callback(null)
+  for (const name of names) {
+    // returnValue being set means we already found what we were looking for, early exit for performance
+    if (returnValue) {
+      break
+    }
+
+    try {
+      const { stderr, stdout } = await exec(`sysctl -n ${name}`)
```

Comment:
i know you didn't write this but this is very unsafe and prone to command injections.  You should use `child_process.execFile('sysctl', ['-n', name])`

Reply:
Here's a decent example https://github.com/nodesecurity/eslint-plugin-security/blob/main/docs/avoid-command-injection-node.md